packages/flutter/lib/src/widgets/notification_listener.dart

```diff
@@ -62,7 +62,7 @@ abstract class Notification {
   @mustCallSuper
   bool visitAncestor(Element element) {
     if (element is StatelessElement) {
-      final StatelessWidget widget = element.widget;
+      final StatelessWidget widget = element.widget as StatelessWidget;
```

Comment:
I believe this cast can also be removed as follows since the code never relies on this being a StatelessWidget:

```suggestion
      final Widget widget = element.widget;
```

Reply:
Ah nice

Reply:
Done

Reply:
actually this whole notification API is sus, I need to look at this more later